[Beaconing] Packaging Network Beaconing Detection (#7418)
elastic/integrations: packages/beaconing/docs/README.md

This package requires a Platinum subscription. Please ensure that you have a Trial, Platinum, or Enterprise subscription before proceeding. This package is licensed under Elastic License 2.0..

## Installation

You can install the Network Beaconing Detection package via **Management > Integrations > Network Beaconing Detection**.

To inspect the installed assets, you can navigate to **Stack Management > Data > Transforms**.

| Transform name            | Purpose| 	Source index  |          Destination index                                                                           |
|---------------------------|--------|----------------|-------------------------------------------------------------------------|
| beaconing.pivot_transform |	Flags beaconing activity in your environment| 	logs-*        |	ml_beaconing_default|

> ## Dashboard

Reviewer: ```suggestion
## Dashboards
```

The **Network Beaconing Detection** has three set of dashboards namely - **Network Beaconing**, which is the main dashboard to monitor beaconing activity, **Beaconing Drilldown** to drilldown into relevant event logs and some statistics related to the beaconing activity, and finally, **Hosts Affected Over Time By Process Name** to monitor the reach of beaconing processes across hosts in your environment.

For the dashboards to work as expected, the following settings need to be configured in Kibana. 
1. Ensure the pivot transform is installed and running.
2. Go to **Management > Stack Management > Kibana > Data Views**. Click on **Create data view** button and enable **Allow hidden and system indices** under the **Show Advanced settings**.
3. Create a data view with the following settings:
    - Index pattern : `ml_beaconing_default`
    - Name: `ml_beaconing_default`
    - Custom data view ID: `ml_beaconing_default`
## Licensing

> Usage in production requires that you have a license key that permits use of machine learning features.

Author: Re-thinking: Guess we can remove this line as this package isn't using any ML features.

> Usage in production requires that you have a license key that permits use of machine learning features.

Reviewer: Ah, true

# Network Beaconing Detection

> The Network Beaconing Detection package consists of a pivot transform required to stand up a framework to identify beaconing activity in your environment. The framework enables significant indicators of compromise (IoCs) for threat hunters and analysts to use as a starting point for an investigation in addition to helping them monitor network traffic for beaconing activities. 

Reviewer 2: ```suggestion
The Network Beaconing Identification package consists of a framework to identify beaconing activity in your environment. The framework surfaces significant indicators of compromise (IoCs) for threat hunters and analysts to use as a starting point for an investigation in addition to helping them monitor network traffic for beaconing activities. 
```

Reviewer 2: Back when we released the package, PM (Sourin) requested that we not claim Detection, but instead position this as a Beaconing Identification capability, so let's retain that messaging.

Author: Sounds good! Changing the package name as well!

> This package requires a Platinum subscription. Please ensure that you have a Trial, Platinum, or Enterprise subscription before proceeding. This package is licensed under Elastic License 2.0..

Author: Thanks for catching! It could work on any subscription level.

## Installation

You can install the Network Beaconing Detection package via **Management > Integrations > Network Beaconing Detection**.

To inspect the installed assets, you can navigate to **Stack Management > Data > Transforms**.

| Transform name            | Purpose| 	Source index  |          Destination index                                                                           |
|---------------------------|--------|----------------|-------------------------------------------------------------------------|

> | beaconing.pivot_transform |	Flags beaconing activity in your environment| 	logs-*        |	ml_beaconing_default|

Reviewer 2: Also if the source index is not space-specific (logs-*), we don't necessarily need to include the space name (default) in the destination index since it might be misleading.

> | beaconing.pivot_transform |	Flags beaconing activity in your environment| 	logs-*        |	ml_beaconing_default|

Author: > Are we able to change this to beaconing_transform? beaconing.pivot_transform looks a bit awkward.

I'm afraid I can't control the naming, it's how it appears after installation via fleet.

## Dashboards

> The **Network Beaconing Detection** has three set of dashboards: 

Reviewer 2: ```suggestion
The **Network Beaconing Detection** has three dashboards: 
```

* **Network Beaconing**: the main dashboard to monitor beaconing activity

> * **Beaconing Drilldown**: drill down into relevant event logs and some statistics related to the beaconing activity

Reviewer 2: ```suggestion
* **Beaconing Drilldown**: Drilldown into relevant event logs and some statistics related to the beaconing activity
```

> * **Hosts Affected Over Time By Process Name**: monitor the reach of beaconing processes across hosts in your environment.

Reviewer 2: ```suggestion
* **Hosts Affected Over Time By Process Name**: Monitor the spread of beaconing processes across hosts in your environment.
```

For the dashboards to work as expected, the following settings need to be configured in Kibana. 
1. Ensure the pivot transform is installed and running.
2. Go to **Management > Stack Management > Kibana > Data Views**. Click on **Create data view** button and enable **Allow hidden and system indices** under the **Show Advanced settings**.
3. Create a data view with the following settings:

>     - Index pattern : `ml_beaconing_default`

Reviewer 2: Same comment as above re: including space in the index name.

The Network Beaconing Identification package consists of a framework to identify beaconing activity in your environment. The framework surfaces significant indicators of compromise (IoCs) for threat hunters and analysts to use as a starting point for an investigation in addition to helping them monitor network traffic for beaconing activities. 
This package is licensed under Elastic License 2.0. 

## Installation

You can install the Network Beaconing Identification package via **Management > Integrations > Network Beaconing Identification**.

To inspect the installed assets, you can navigate to **Stack Management > Data > Transforms**.

| Transform name            | Purpose| 	Source index  | Destination index       | Alias |
|---------------------------|--------|----------------|-------------------------|------------|
| beaconing.pivot_transform |	Flags beaconing activity in your environment| 	logs-*        | 	ml_beaconing-[version] | ml_beaconing.all |

For additional information on the transform's inner workings and the signals it generates, refer to [this blog post](https://www.elastic.co/security-labs/identifying-beaconing-malware-using-elastic).

> **Note**: When querying the destination index to enquire about beaconing activities, we advise using the alias for the destination index (`ml_beaconing.all`). In the event that the underlying package is upgraded, the alias will aid in maintaining the previous findings.

Reviewer 2: ```suggestion
**Note**: When querying the destination index to enquire about beaconing activities, we advise using the alias for the destination index (`ml_beaconing.all`). In the event that the underlying package is upgraded, the alias will aid in maintaining the previous findings.

```

## Dashboards

The **Network Beaconing Identification** has three dashboards: 
* **Network Beaconing**: The main dashboard to monitor beaconing activity
* **Beaconing Drilldown**: Drilldown into relevant event logs and some statistics related to the beaconing activity
* **Hosts Affected Over Time By Process Name**: Monitor the spread of beaconing processes across hosts in your environment

For the dashboards to work as expected, the following settings need to be configured in Kibana. 
1. Ensure the pivot transform is installed and running.
2. Go to **Management > Stack Management > Kibana > Data Views**. Click on **Create data view** button and enable **Allow hidden and system indices** under the **Show Advanced settings**.
3. Create a data view with the following settings:
    - Index pattern : `ml_beaconing.all`
    - Name: `ml_beaconing`

>     - Custom data view ID: `ml_beaconing`

Reviewer 2: ```suggestion
    - Custom data view ID: `ml_beaconing`
    
## Enabling detection rules

You can also enable detection rules to alert on Network Beaconing activity in your environment. These rules are available as part of the Detection Engine, and can be found using the tag `Use Case: C2 Beaconing Detection`. See this [documentation](https://www.elastic.co/guide/en/security/current/prebuilt-rules-management.html#load-prebuilt-rules) for more information on importing and enabling the rules.
```

Author: We should hold off on adding a note about the rules until the detection-rules PR is merged. Given that, I won't pursue this suggestion at the moment.
